crates/deltalake-core/src/kernel/actions/schemas.rs

```diff
@@ -9,236 +9,241 @@ lazy_static! {
+                MapType::new(
+                    DataType::STRING,
+                    DataType::STRING,
                     true,
-                ))),
+                ),
                 false,
             ),
-        ]))),
+        ]),
         true,
     );
     // https://github.com/delta-io/delta/blob/master/PROTOCOL.md#protocol-evolution
     static ref PROTOCOL_FIELD: StructField = StructField::new(
         "protocol",
-        DataType::Struct(Box::new(StructType::new(vec![
-            StructField::new("minReaderVersion", DataType::integer(), false),
-            StructField::new("minWriterVersion", DataType::integer(), false),
+        StructType::new(vec![
+            StructField::new("minReaderVersion", DataType::INTEGER, true),
+            StructField::new("minWriterVersion", DataType::INTEGER, true),
             StructField::new(
                 "readerFeatures",
-                DataType::Array(Box::new(ArrayType::new(DataType::string(), false))),
+                ArrayType::new(DataType::STRING, true),
                 true,
             ),
             StructField::new(
                 "writerFeatures",
-                DataType::Array(Box::new(ArrayType::new(DataType::string(), false))),
+                ArrayType::new(DataType::STRING, true),
                 true,
             ),
-        ]))),
+        ]),
         true,
     );
     // https://github.com/delta-io/delta/blob/master/PROTOCOL.md#commit-provenance-information
     static ref COMMIT_INFO_FIELD: StructField = StructField::new(
         "commitInfo",
-        DataType::Struct(Box::new(StructType::new(vec![
-            StructField::new("timestamp", DataType::timestamp(), false),
-            StructField::new("operation", DataType::string(), false),
-            StructField::new("isolationLevel", DataType::string(), true),
-            StructField::new("isBlindAppend", DataType::boolean(), true),
-            StructField::new("txnId", DataType::string(), true),
-            StructField::new("readVersion", DataType::long(), true),
+        StructType::new(vec![
+            StructField::new("timestamp", DataType::LONG, false),
+            StructField::new("operation", DataType::STRING, false),
+            StructField::new("isolationLevel", DataType::STRING, true),
+            StructField::new("isBlindAppend", DataType::BOOLEAN, true),
+            StructField::new("txnId", DataType::STRING, true),
+            StructField::new("readVersion", DataType::LONG, true),
             StructField::new(
                 "operationParameters",
-                DataType::Map(Box::new(MapType::new(
-                    DataType::string(),
-                    DataType::string(),
+                MapType::new(
+                    DataType::STRING,
+                    DataType::STRING,
                     true,
-                ))),
+                ),
                 true,
             ),
             StructField::new(
                 "operationMetrics",
```

Comment:
This wouldn't be valid according to the protocol and our current operations since we also write operationMetrics as   MapType::new(
                    DataType::STRING,
                    DataType::INTEGER)